uncore_csv_json.py

```diff
@@ -214,10 +214,7 @@ def is_deprecated(j):
             verboseprint("Both event", name, "and its new name", newname, "are supported", file=sys.stderr)
 
     if all_events:
-        def skip(j):
-            return "Deprecated" in j and j["Deprecated"] == "1"
-
-        jl += [update(events[x]) for x in sorted(events.keys()) if not skip(events[x]) and not x in added]
+        jl += [update(events[x]) for x in sorted(events.keys()) if not x in added]
```

Comment:
It's better to add an option for "Deprecated" events instead to add them in default.

Reply:
For core events there is no option, so this brings about greater consistency. Such an option is unnecessary as perf has a deprecated flag and hides deprecated events by default.

Reply:
Add an option maybe is useful, when the perf "deprecated flag" function has a problem, we can easy to generate the event list without "deprecated" events and quickly backtrace the issue.

Reply:
It could. I don't see why it'd be here and specific to uncore events. We could write a new "strip deprecated" command for this, it would just json into json and drop any dictionaries containing 'Deprecated' : '1'.

Reply:
The latest commit adds a command:
remove-deprecated-json.py
that will remove "Deprecated":"1" events and write them to stdout or a specified filename.